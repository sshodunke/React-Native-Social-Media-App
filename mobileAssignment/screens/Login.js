import React, { Component } from 'react'; 
import { Text, View, StyleSheet, KeyboardAvoidingView, TouchableOpacity, Alert } from 'react-native';
import AsyncStorage from '@react-native-community/async-storage'
import { TextInput } from 'react-native-paper';
import axios from 'axios';

const options = {
    headers: {'Accept': 'application/json', 'Content-Type': 'application/json',}
};

class Login extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            token: '',
        }
    }

    sendLoginRequest() {
       return axios.post('http://10.0.2.2:3333/api/v0.0.5/login', {
            email: this.state.email,
            password: this.state.password
       }, options)

       .then((response) => {
           AsyncStorage.setItem('token', toString(response.data))
           this.props.navigation.navigate('Home')
       })

       .catch(function(error) {
           console.log(error)
           Alert.alert('Incorrect Email/Password')
       })
    }

    async retrieveToken() {
        try {
            const value = await AsyncStorage.getItem('token');
            if (value !== null) {
                 this.token = value
            }
        } catch (error) {
        }
    };

    componentDidMount() {
        this.retrieveToken();
    }

    render() {
        return (
            <KeyboardAvoidingView behavior='padding' style={styles.wrapper}>
                
                <View style={styles.container}>

                    <View style={styles.header}>
                        <Text style={styles.header}>Chittr</Text>
                    </View>

                    <View>
                        <TextInput
                            onSubmitEditing={() => this.passwordInput.focus()}
                            style={styles.textInput}
                            placeholder='Email'
                            onChangeText={(email) => this.setState({email})}
                            underlineColorAndroid='transparent'/>

                        <TextInput
                            secureTextEntry={true}
                            ref={(input) => this.passwordInput = input} 
                            style={styles.textInput}
                            placeholder='Password'
                            onChangeText={(password) => this.setState({password})}
                            underlineColorAndroid='transparent'/>

                        <TouchableOpacity
                            style={styles.btn}
                            onPress={() => this.sendLoginRequest()}>
                            <Text>Login</Text>
                        </TouchableOpacity>

                        <Text onPress={() => this.props.navigation.navigate('SignUp')} style={styles.register}>Register a new account.</Text>
                    </View>

                </View>
                
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    wrapper: {
        flex: 1,
        backgroundColor: '#5c6bc0'
    },
    container: {
        flex: 1,
        paddingHorizontal: 30,
    },
    header: {
        fontSize: 24,
        fontFamily: 'Avenir Next',
        color: '#ffffff',
        marginTop: 48,
        marginBottom: 60,
        alignItems: "center",
        justifyContent: "center",
    },
    textInput: {
        marginBottom: 20,
        alignSelf: "stretch",
    },
    text: {
        fontFamily: 'Avenir Next',
        color: '#ffffff'
    },
    btn: {
        backgroundColor: '#8e99f3',
        fontSize: 16,
        borderRadius: 4,
        paddingVertical: 12,
        marginTop: 32,
        alignItems: "center",
        justifyContent: "center",
    },
    register: {
        marginTop: 24,
        fontWeight: '500',
        fontSize: 14,
        color: '#ffffff',
        textAlign: "center",
    }
})

export default Login;