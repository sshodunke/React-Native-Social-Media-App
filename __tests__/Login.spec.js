import 'react-native';
import React from 'react';
import Login from '../screens/Login'

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

describe('Login Component', () => {
  let store
  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    })
  })

  it('Login component should render and match snapshot', () => {
    expect(renderer.create(
      <Provider store={store}>
        <Login/>
      </Provider>
    ).toJSON()).toMatchSnapshot()

  })
})