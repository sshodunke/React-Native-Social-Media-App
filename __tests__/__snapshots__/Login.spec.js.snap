// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Login component should render and match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#5c6bc0",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "color": "#ffffff",
          "fontFamily": "Avenir Next",
          "fontSize": 24,
          "justifyContent": "center",
          "marginBottom": 60,
          "marginTop": 48,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#ffffff",
            "fontFamily": "Avenir Next",
            "fontSize": 24,
            "justifyContent": "center",
            "marginBottom": 60,
            "marginTop": 48,
          }
        }
      >
        Chittr
      </Text>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "alignSelf": "stretch",
              "marginBottom": 20,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Email"
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "paddingHorizontal": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "alignSelf": "stretch",
              "marginBottom": 20,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            secureTextEntry={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "paddingHorizontal": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#8e99f3",
            "borderRadius": 4,
            "fontSize": 16,
            "justifyContent": "center",
            "marginTop": 32,
            "opacity": 1,
            "paddingVertical": 12,
          }
        }
      >
        <Text>
          Login
        </Text>
      </View>
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "500",
            "marginTop": 24,
            "textAlign": "center",
          }
        }
      >
        Register a new account.
      </Text>
    </View>
  </View>
</View>
`;
